tests/test_models.py

```diff
@@ -1197,36 +1160,37 @@ def test_string_conversion(self, method):
 @mark.django_db
 class TestEnterpriseCustomerPluginConfiguration(unittest.TestCase):
     """
-    Tests of the EnterpriseCustomerPluginConfiguration abstract model.
+    Tests of the ``EnterpriseCustomerPluginConfiguration`` base model.
     """
 
     def setUp(self):
-        self.abstract_base = EnterpriseCustomerPluginConfiguration()
+        self.enterprise_customer = EnterpriseCustomerFactory()
+        self.config = EnterpriseCustomerPluginConfiguration(enterprise_customer=self.enterprise_customer)
         super(TestEnterpriseCustomerPluginConfiguration, self).setUp()
 
     def test_channel_code_raises(self):
         with raises(NotImplementedError):
-            self.abstract_base.channel_code()
+            self.config.channel_code()
 
-    def test_get_learner_data_record_raises(self):
-        with raises(NotImplementedError):
-            self.abstract_base.get_learner_data_record(mock.Mock())
+    @mock.patch('integrated_channels.integrated_channel.models.LearnerExporter')
+    def test_get_learner_data_exporter(self, mock_learner_exporter):
+        mock_learner_exporter.return_value = 'mock_learner_exporter'
+        assert self.config.get_learner_data_exporter(None) == 'mock_learner_exporter'
 
-    def test_get_learner_data_exporter_raises(self):
-        with raises(NotImplementedError):
-            self.abstract_base.get_learner_data_exporter(mock.Mock())
-
-    def test_get_learner_data_transmitter_raises(self):
-        with raises(NotImplementedError):
-            self.abstract_base.get_learner_data_transmitter()
+    @mock.patch('integrated_channels.integrated_channel.models.LearnerTransmitter')
+    def test_get_learner_data_transmitter_raises(self, mock_learner_transmitter):
+        mock_learner_transmitter.return_value = 'mock_learner_transmitter'
+        assert self.config.get_learner_data_transmitter() == 'mock_learner_transmitter'
 
-    def test_get_course_data_exporter_raises(self):
-        with raises(NotImplementedError):
-            self.abstract_base.get_course_data_exporter(None)
+    @mock.patch('integrated_channels.integrated_channel.models.CourseExporter')
+    def test_get_course_data_exporter_raises(self, mock_course_exporter):
+        mock_course_exporter.return_value = 'mock_course_exporter'
+        assert self.config.get_course_data_exporter(None) == 'mock_course_exporter'
 
-    def test_get_course_data_transmitter_raises(self):
-        with raises(NotImplementedError):
-            self.abstract_base.get_course_data_transmitter()
+    @mock.patch('integrated_channels.integrated_channel.models.CourseTransmitter')
+    def test_get_course_data_transmitter_raises(self, mock_course_transmitter):
+        mock_course_transmitter.return_value = 'mock_course_transmitter'
+        assert self.config.get_course_data_transmitter() == 'mock_course_transmitter'
```

Comment:
@UmanShahzad Docstrings for the four tests above would be nice.

Reply:
Will address this when we begin working on the tests ticket (there's definitely work that has to be done for both SAPSF/Degreed tests, and not just the latter).

Reply:
:+1: